File: csharp/ItemModels/AgedBrie.cs
using System;
namespace csharp.ItemModels
{
	public class AgedBrie : Common
	{
        public AgedBrie(Item item) : base ( item)
        {
            _item = item;
        }

        public override void UpdateQuality() // "Aged Brie"（陈年布利奶酪）的品质`Quality`会随着时间推移而提高
        {
            int vibrantValue = 1; // 震盪數值
            if (_item.SellIn <= 0) // 一旦销售期限过期，品质`Quality`会以双倍速度加速增加 or 下降
                vibrantValue *= 2;
            _item.Quality += vibrantValue;
            QuilityAdjust();
        }
    }
}

File: js-jest/src/item/agedBrie.js
const Item = require('./item');

class AgedBrie extends Item {
    constructor(sellIn, quality) {
        super('Aged Brie', sellIn, quality);
    }

    updateItem() {
        this.sellIn -= 1;
        // "Aged Brie"（陈年布利奶酪）的品质`Quality`会随着时间推移而提高
        // 物品的品质`Quality`永远不会超过50
        this.quality = Math.min(50, (this.sellIn > 0 ? this.quality + 1 : this.quality + 2));
    }
}

module.exports = AgedBrie;
